compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -5690,7 +5690,7 @@ private BErrorType createErrorType(BType lhsType, BType rhsType, BType detailTyp
         BErrorType lhsErrorType = (BErrorType) lhsType;
         BErrorType rhsErrorType = (BErrorType) rhsType;
 
-        BErrorType errorType = createErrorType(detailType, lhsType.flags, env);
+        BErrorType errorType = createErrorType(detailType, lhsType.flags | Flags.PUBLIC, env);
         errorType.tsymbol.flags |= rhsType.flags;
```

Comment:
Any idea why we need to set the flags of the `rhs` type only for the type symbol of the error type? If we don't need this, we can remove it. If we do need to assign the flag to the type symbol, then those flags should be assigned to the type as well. So, we can pass these flags of the `rhs` type to the `createErrorType` function.
```
BErrorType errorType = createErrorType(detailType, lhsType.flags| lhsType.flags | Flags.PUBLIC, env);
```

Reply:
As discussed in offline, we have to have both RHS and LHS flags for the error intersections